src/TimeSpans.jl

```diff
@@ -109,6 +110,10 @@ Return the inclusive lower bound of `span` as a `Nanosecond` value.
 """
 start(span::TimeSpan) = span.start
 start(t::Period) = convert(Nanosecond, t)
+function start(x::NamedTuple)
+    x.start isa Period || throw(ArgumentError("The field `start` must be a `Period` type"))
+    Nanosecond(x.start)
+end
```

Comment:
```suggestion

function start(x::NamedTuple)
    istimespan(x) || throw(ArgumentError("input is not a valid timespan"))
    return Nanosecond(x.start)
end
```
We probably don't want to support `start` for `NamedTuple`s that don't fully comply with the timespan interface.